File: Master/1st Course/Spring semester/Moderno programmesanas valodu praktikums/Lab_2/Employee.java
package lab2.concepts;

public class Employee extends Human implements Comparable<Employee> {
    private double salary;
    
    public Employee() {
        super();
    }
    
    public Employee(String name, String surname, double salary) {
        super(name, surname);
        this.salary = salary;
    }
    
    public Employee(Employee employee) {
        super(employee);
        this.salary = employee.salary;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    @Override
    public String toString() {
        return super.toString() + ", alga: " + salary;
    }

    @Override
    public int compareTo(Employee e) {
        return this.salary > e.salary ? 1 :
            this.salary < e.salary ? -1 : 0;
    }
}

File: Master/1st Course/Spring semester/Moderno programmesanas valodu praktikums/Lab_8/Employee.cs
using System;

namespace Lab8 {

    public class Employee : Human, IComparable<Employee> {

        public double Salary { get; set; }

        public Employee() { }

        public Employee(string name, string surname, double salary) 
            : base(name, surname) {
            Salary = salary;
        }

        public Employee(Employee employee) : base(employee) {
            Salary = employee.Salary;
        }

        public int CompareTo(Employee other) {
            return Salary == other.Salary ? 0 : Salary > other.Salary ? 1 : -1;
        }

        public override string ToString() {
            return base.ToString() + string.Format(" alga: {0}", Salary);
        }
    }
}
